Content.Server/Store/Systems/StoreSystem.cs

```diff
@@ -36,12 +36,14 @@ public override void Initialize()
 
         InitializeUi();
         InitializeCommand();
+        InitializeRefund();
     }
 
     private void OnMapInit(EntityUid uid, StoreComponent component, MapInitEvent args)
     {
         RefreshAllListings(component);
         InitializeFromPreset(component.Preset, uid, component);
+        component.StartingMap = Transform(uid).MapID;
```

Comment:
As above you don't need to be doing this and it's missing edge cases anyway.

Reply:
How would I set the starting map?

Reply:
Set it on MapInit.